src/mnode/mnode-pastel.cpp

```diff
@@ -854,24 +921,31 @@ bool CArtSellTicket::IsValid(bool preReg, int depth) const
             existingTicket.m_txid == m_txid) // if this ticket is already in the DB
             ticketFound = true;
     }
-    
-    //1. check PastelID in this ticket matches PastelID in the referred ticket (Activation or Trade)
-    //2. Verify the art is not already sold
-    auto existingTradeTickets = CArtTradeTicket::FindAllTicketByArtTnxID(artTnxId);
-    auto soldCopies = existingTradeTickets.size();
-    auto existingSellTickets = CArtSellTicket::FindAllTicketByArtTnxID(artTnxId);
-    auto sellTicketsNumber = existingSellTickets.size();
-    auto totalCopies = 0;
+
+    // Check PastelID in this ticket matches PastelID in the referred ticket (Activation or Trade)
+    auto totalCopies{0};
```

Comment:
should be size_t type.
I think by default c++ will make for 0 -> int type.
Then  for (soldCopies >= totalCopies)  comparison compiler will give a warning: signed and unsigned data types comparison

Reply:
correct

Reply:
fixed